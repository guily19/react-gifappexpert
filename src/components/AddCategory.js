import React, {useState} from 'react'
import PropTypes from 'prop-types'

const AddCategory = (props) => {

  const { setCategories } = props;
  const [inputValue, setInputValue] = useState('');

  const handlOnSubmitForm = (e) => {
    e.preventDefault();
    if ( inputValue.trim().length > 2 ){
      setCategories((prev) => {
        return [inputValue, ...prev]
      });
      setInputValue('');
    }
  }

  const handlOnChangeInput = (e) => {
    setInputValue(e.target.value);
  }
  return (
    <form onSubmit={handlOnSubmitForm}>
      <input
        type="text"
        value={inputValue}
        onChange={handlOnChangeInput}
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;