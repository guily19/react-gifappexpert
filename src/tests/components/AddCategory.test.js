import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('AddCategory', () => {
  it('match the snapshot', () =>{
    const setCategories = jest.fn();
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the input content', () =>{
    const setCategories = jest.fn();
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);
    const input = wrapper.find('input');
    input.simulate('change', {target: {value: 'text'}});
    const pElement = wrapper.find('p').text().trim();
    expect(pElement).toBe('text');
  });

  it(' not post on submit', () => {
    const setCategories = jest.fn();
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);
    wrapper.find('form').simulate('submit', { preventDefault : () => {}});
    expect( setCategories ).not.toHaveBeenCalled();
  });

  it('post on submit and clean the input', () => {
    const setCategories = jest.fn();
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);
    const input = wrapper.find('input');
    input.simulate('change', {target: {value: 'text'}});
    wrapper.find('form').simulate('submit', { preventDefault : () => {}});
    expect( setCategories ).toHaveBeenCalled();
    expect( setCategories ).toHaveBeenCalledTimes(1);
    expect( setCategories ).toHaveBeenCalledWith( expect.any(Function));
    expect(input.text()).toBe('');
  });
})