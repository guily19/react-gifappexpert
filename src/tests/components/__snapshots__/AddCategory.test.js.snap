// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory match the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
